lib/productions/type.js

```diff
@@ -91,6 +91,7 @@ function union_type(tokeniser, type) {
   while (true) {
     const typ = type_with_extended_attributes(tokeniser) || tokeniser.error("No type after open parenthesis or 'or' in union type");
     if (typ.idlType === "any") tokeniser.error("Type `any` cannot be included in a union type");
+    if (typ.generic === "Promise") tokeniser.error("Type `Promise` cannot be included in a union type");
```

Comment:
This is already good!

But could we move this two lines (line 93 and 94) before line 92, and change them into `if (tokeniser.probe("any"))` and `if (tokeniser.probe("Promise"))` instead of current one? That would create a better error message.